website/docs/r/nsxt_edgegateway.html.markdown

```diff
@@ -223,12 +272,29 @@ definition at provider level.
 * `description` - (Optional) A unique name for the edge gateway.
 * `external_network_id` - (Required) An external network ID. **Note:** Data source [vcd_external_network_v2](/providers/vmware/vcd/latest/docs/data-sources/external_network_v2)
 can be used to lookup ID by name.
-* `subnet` - (Required) One or more [subnets](#edgegateway-subnet) defined for edge gateway.
 * `edge_cluster_id` - (Optional) Specific Edge Cluster ID if required
 * `dedicate_external_network` - (Optional) Dedicating the External Network will enable Route Advertisement for this Edge Gateway. Default `false`.
 
+* `subnet` - (Optional) One or more [subnets](#edgegateway-subnet) defined for edge gateway. One of
+  `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_subnet` - (Optional, *v3.9+*) One or more [subnets](#edgegateway-auto-subnet) defined for
+  Edge Gateway. One of `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_allocated_subnet` - (*v3.9+*) One or more [subnets](#edgegateway-auto-allocated-subnet)
+  defined for Edge Gateway. One of `subnet`, `auto_subnet` or `auto_allocated_subnet` is
+  **required**.
+* `total_allocated_ip_count` - (Optional, *v3.9+*) Required with `auto_subnet`
+
+~> Starting with v3.9 of this provider, NSX-T Edge Gateways can allocate IP addresses by using
+different strategies: Manual IP allocation (`subnet`), automatic IP allocations in any of defined
+subnets (`auto_subnet` with `total_allocated_ip_count`), automatic IP allocations per defined subnet
+(`auto_allocated_subnet`). One of these is **required**. Different set definition structures are
```

Comment:
What if we rephrased this to something bulleted and expanded like this:

> Starting with v3.9 of provider, NSX-T Edge Gateways can allocate IP addresses by using
> three different strategies:
> - Manual IP allocation (`subnet`) - have multiple start-end IP ranges (`allocated_ips` blocks) in a subnet
> - Automatic IP allocations in any of defined subnets (`auto_subnet` with `total_allocated_ip_count`) - have all IPs in a subnet, but must limit selection with a global `total_allocated_ip_count` setting
> - Automatic IP allocations per subnet
> (`auto_allocated_subnet`) - limit IP count in each subnet with `allocated_ip_count`, but *must not* limit with global `total_allocated_ip_count` setting

Reply:
I have revisited it, but there is a limitation of highlighting syntax. I am not able to add bullet points to the note itself (see the screenshot of how it renders).
I can do multiple notes with each option only.

<img width="715" alt="image" src="https://user-images.githubusercontent.com/15804230/224916294-fe53c5fb-5ca3-4b13-bb47-c1e0452b2ec7.png">

Reply:
I see. I think we need to do some tidying up of all the subnet section. When you look at them right now it's a bit unclear which section relates to wich argument:

![Screenshot 2023-03-14 at 10 26 58](https://user-images.githubusercontent.com/11330689/224941017-7eb53cff-247d-4ef9-8b02-8d3c160f649b.png)

Reply:
Thanks for the overhaul!